libs/community/tests/integration_tests/vectorstores/test_aerospike.py

```diff
@@ -0,0 +1,838 @@
+            embedder,
+            client=client,
+            namespace=TEST_NAMESPACE,
+            index_name=index_name,
+            ids=["1", "2", "3", "4", "5", "6", "7"],
+            set_name=set_name,
+        )
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=3, fetch_k=3
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"),
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "baz"
+
+        # if fetch_k < k, then the output will be less than k
+        mmr_output = aerospike.max_marginal_relevance_search_by_vector(
+            embedder.embed_query("foo"), index_name=index_name, k=3, fetch_k=2
+        )
+        assert len(mmr_output) == 2
+
+    def test_max_marginal_relevance_search(
+        self, aerospike: Aerospike, admin_client: Any
+    ) -> None:
+        """Test max marginal relevance search."""
+
+        index_name = set_name = get_func_name()
+        admin_client.index_create(
+            namespace=TEST_NAMESPACE,
+            sets=set_name,
+            name=index_name,
+            vector_field=VECTOR_KEY,
+            dimensions=10,
+        )
+        aerospike.add_texts(
+            ["foo", "bar", "baz", "bay", "bax", "baw", "bav"],
+            ids=["1", "2", "3", "4", "5", "6", "7"],
+            index_name=index_name,
+            set_name=set_name,
+        )
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=3, fetch_k=3
+        )
+        sim_output = aerospike.similarity_search("foo", index_name=index_name, k=3)
+
+        assert len(mmr_output) == 3
+        assert mmr_output == sim_output
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo", index_name=index_name, k=2, fetch_k=3
+        )
+
+        assert len(mmr_output) == 2
+        assert mmr_output[0].page_content == "foo"
+        assert mmr_output[1].page_content == "bar"
+
+        mmr_output = aerospike.max_marginal_relevance_search(
+            "foo",
+            index_name=index_name,
+            k=2,
+            fetch_k=3,
+            lambda_mult=0.1,  # more diversity
+        )
```

Comment:
**Security Issue**: The lambda_mult parameter is set to 0.1 to increase diversity in the max marginal relevance search, but this might lead to less relevant results being included in the output. This could potentially expose sensitive or less relevant data to users if not handled correctly. <br> **Fix**: Adjust the lambda_mult parameter to a higher value to ensure that the results are more relevant and reduce the risk of exposing less relevant or sensitive data. <br> **Code Suggestion**: 
 ```
 -            lambda_mult=0.1,  # more diversity
 +            lambda_mult=0.5,  # balanced diversity
 ```